handlers/dimensions.go

```diff
@@ -247,53 +276,56 @@ func (f *Filter) AddList(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	if _, ok := req.Form["All times"]; ok {
-		f.AddAll(w, req)
+	redirectURL := fmt.Sprintf("/filters/%s/dimensions", filterID)
+
+	if len(req.Form["add-all"]) > 0 {
+		redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s?selectorType=list", filterID, name)
+		f.addAll(w, req, redirectURL)
 		return
 	}
 
-	values, labelIDMap, err := f.getDimensionValues(filterID, name)
-	if err != nil {
-		log.ErrorR(req, err, nil)
-		w.WriteHeader(http.StatusInternalServerError)
+	if len(req.Form["remove-all"]) > 0 {
+		redirectURL = fmt.Sprintf("/filters/%s/dimensions/%s/remove-all?selectorType=list", filterID, name)
+		http.Redirect(w, req, redirectURL, 302)
 		return
 	}
 
-	if name == "time" {
-		dats, err := dates.ConvertToReadable(values)
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	// concurrently remove any fields that have been deselected
+	go func() {
+		opts, err := f.fc.GetDimensionOptions(filterID, name)
```

Comment:
There are several clients for each api so couldn't just use Client (see clients.go for all clients)